plugins/module_utils/network/ios/facts/route_maps/route_maps.py

```diff
@@ -0,0 +1,263 @@
+    Route_mapsTemplate,
+)
+from ansible_collections.cisco.ios.plugins.module_utils.network.ios.argspec.route_maps.route_maps import (
+    Route_mapsArgs,
+)
+
+
+class Route_mapsFacts(object):
+    """ The cisco.ios route_maps facts class
+    """
+
+    def __init__(self, module, subspec="config", options="options"):
+        self._module = module
+        self.argument_spec = Route_mapsArgs.argument_spec
+        spec = deepcopy(self.argument_spec)
+        if subspec:
+            if options:
+                facts_argument_spec = spec[subspec][options]
+            else:
+                facts_argument_spec = spec[subspec]
+        else:
+            facts_argument_spec = spec
+
+        self.generated_spec = utils.generate_dict(facts_argument_spec)
+
+    def get_route_maps_data(self, connection):
+        return connection.get("sh running-config | section ^route-map")
+
+    def populate_facts(self, connection, ansible_facts, data=None):
+        """ Populate the facts for Route_maps network resource
+
+        :param connection: the device connection
+        :param ansible_facts: Facts dictionary
+        :param data: previously collected conf
+
+        :rtype: dictionary
+        :returns: facts
+        """
+        facts = {}
+        objs = []
+
+        if not data:
+            data = self.get_route_maps_data(connection)
+        # parse native config using the Route_maps template
+        route_maps_parser = Route_mapsTemplate(lines=data.splitlines())
+        objs = route_maps_parser.parse()
+        # objs = list(route_maps_parser.parse().values())
+        final_objs = []
+        if objs:
+            for k, v in iteritems(objs):
+                temp_dict = {}
+                temp_dict["entries"] = []
+                for key, val in iteritems(v):
+                    if key == "route_map":
+                        temp_dict.update({"route_map": val})
+                        continue
+                    if val.get("entries"):
+                        if val["entries"].get("match"):
+                            if val["entries"]["match"].get("local_preference"):
+                                local_preference_val = [
+                                    each
+                                    for each in val["entries"]["match"][
+                                        "local_preference"
+                                    ]["value"]
+                                    if each
+                                ]
+                                if local_preference_val:
+                                    if isinstance(
+                                        local_preference_val[0], str
+                                    ):
+                                        val["entries"]["match"][
+                                            "local_preference"
+                                        ]["value"] = local_preference_val[
```

Comment:
Is this logic used for splitting a string into a list? If so, this can be done in the parsers. Please refer to https://github.com/ansible-collections/cisco.nxos/pull/283/files#diff-46ccf244c92a3702bab70313ef00867ec6c18922c6d1c34d9c82abbfb167b33dR351.

Reply:
+1